### PR TITLE
Upgrade go version to 1.18

### PR DESCRIPTION
**Pull request by wansir**
### What type of PR is this?

Upgrade go version to 1.18.

### Which issue(s) this PR fixes:

Fixes #

### Special notes for reviewers:
```
```

### Does this PR introduced a user-facing change?

```release-note
None
```

### Additional documentation, usage docs, etc.:

```docs

```

### CODE REVIEW

File: LICENSES/LICENSE [hunk 1]
```diff
@@ -0,0 +1,207 @@
+================================================================================
+= KubeSphere licensed under: =
```
Comment (iawia002) [inline LICENSES/LICENSE:2]: This format looks weird, why is there a `:`

Comment (wansir) [reply LICENSES/LICENSE:2]: The script is copied from https://github.com/kubernetes/kubernetes/blob/master/hack/update-vendor-licenses.sh. The license directory is helpful for us.

Comment (iawia002) [reply LICENSES/LICENSE:2]: Got it.

File: hack/update-vendor-licenses.sh [hunk 2]
```diff
@@ -155,55 +155,80 @@ fi
...
+mkdir -p "${LICENSES_DIR}"
+
+# The tmp directory to save all the LICENSE files, will move to LICENSES_DIR
+TMP_LICENSES_DIR="${KUBE_TEMP}/LICENSES.DIR.$$"
+mkdir -p "${TMP_LICENSES_DIR}"
+
 DEPS_DIR="vendor"
 declare -Ag CONTENT
 
 # Put the K8S LICENSE on top
```
Comment (xyz-li) [inline hack/update-vendor-licenses.sh]: `Put the KubeSphere LICENSE on top`

Comment (wansir) [reply hack/update-vendor-licenses.sh]: Updated

**Pull request merged**